CMakeLists.txt

```diff
@@ -1,73 +1,71 @@
+    endif()
 endif()
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+# project version
 include(FunctionReadVersion)
+read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" QBT_VER_MAJOR QBT_VER_MINOR QBT_VER_BUGFIX QBT_VER_BUILD QBT_VER_STATUS)
+message(STATUS "Project version is: ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD} (${QBT_VER_STATUS})")
 
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_BUGFIX VER_BUILD VER_STATUS)
-# message(STATUS "Project version is: ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD} (${VER_STATUS})")
-
-project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
+project(qBittorrent VERSION ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD})
 
 # check for invalid compiler version/CXX standard as early as possible
+# ideally, we would also be able to check whether or not libtorrent was compiled
+# with the same standard from imported target properties
 include(FunctionQbtCXXCompilerAndModeCheck)
 qbt_minimum_cxx_mode_check(14)
-message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
-            "Make sure libtorrent was built with the same C++ mode for ABI compatibility.")
-
-set(PROJECT_VERSION "${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}")
-
-if (NOT VER_BUILD EQUAL 0)
-    set(PROJECT_VERSION "${PROJECT_VERSION}.${VER_BUILD}")
-endif()
-
-set(PROJECT_VERSION "${PROJECT_VERSION}${VER_STATUS}")
-
-add_definitions(-DQBT_VERSION_MAJOR=${VER_MAJOR})
-add_definitions(-DQBT_VERSION_MINOR=${VER_MINOR})
-add_definitions(-DQBT_VERSION_BUGFIX=${VER_BUGFIX})
-add_definitions(-DQBT_VERSION_BUILD=${VER_BUILD})
-
-add_definitions(-DQBT_VERSION="v${PROJECT_VERSION}")
-add_definitions(-DQBT_VERSION_2="${PROJECT_VERSION}")
-
-include(GNUInstallDirs)
-include(FeatureSummary)
 
 # version requirements
-set(requiredBoostVersion 1.40)
+set(requiredBoostVersion 1.60)
 set(requiredQtVersion 5.9.0)
-set(requiredOpensslVersion 1.0)
-set(requiredLibtorrentVersion 1.1.10)
+set(requiredOpenSSLVersion 1.0)
+set(requiredLibtorrentVersion 1.2.0)
+set(requiredZlibVersion 1.2.5.2)
 
-if (WIN32)
-    include(winconf)
+# features (some are platform-specific)
+include(FeatureSummary)
+include(FeatureOptionsSetup)
+feature_option(COUNTRIES_RESOLUTION "Enable resolving peers IP addresses to countries" ON)
+feature_option(STACKTRACE "Enable stacktraces" ON)
+feature_option(GUI "Build GUI application" ON)
+feature_option(WEBUI "Enables built-in HTTP server for headless use" ON)
+feature_option(VERBOSE_CONFIGURE "Show information about PACKAGES_FOUND and PACKAGES_NOT_FOUND in the configure output (only useful for debugging the CMake build scripts)" OFF)
+if (UNIX)
+    feature_option_dependent(DBUS
+        "Enables features such as notification support and optional power-management features on Linux via D-Bus"
```

Comment:
I don't think power-management should be optional in case when DBUS is available.

Reply:
@glassez 
Currently it's optional, but turned on by default (as long as DBUS is also ON, if DBUS is OFF, it's always turned off). Why not give the users the possibility to turn it off? Everything else seems to be in place for that.

Reply:
What's the point to make such subfeatures to be optional? What's benefit in disabling it in compile time?

Reply:
@glassez 
Because this is what the script was doing before, but through "spooky action at a distance" - I just made it a user-configurable option.

https://github.com/qbittorrent/qBittorrent/blob/fb7774705d379a07440586762c6fc057654d5f67/src/gui/CMakeLists.txt#L213

Do you think in this case the choice does not make sense at all?

Reply:
>I just made it a user-configurable option.
Do you think in this case the choice does not make sense at all?

I think it doesn't make sense. This option should be always enabled when all necessary stuff (e.g. DBUS) is available.

Here is my general opinion.
You need to distinguish between different types of "optional" features.
Some of them are optional because it makes sense to disable them at the user's choice (for example, GUI mode).
Some others are optional because they require some additional dependencies, but we can do without them if these dependencies cannot be satisfied for some reason (e.g. "power management").

>Because this is what the script was doing before, but through "spooky action at a distance" 

You must be mistaken. It simply disabled the feature if there were unsatisfied dependencies.

Reply:
@glassez Understood, fixed.